src/luacov-console.lua

```diff
@@ -27,6 +27,14 @@ local function colored_print(color, text)
     print(start_color .. text .. end_color)
 end
 
+local function colored_print_if_pattern(color, text, patterns)
```

Comment:
According to this new name, maybe we should introduce an optimization that if the patterns are nil, we call colored_print directly. And for cases without patterns, we can simply pass a nil as the patterns.

Reply:
it is an optimisation true and it passed my mind as well. 
but then thought to go with less cyclomatic complexity instead.
do you really want that if-statement?

Reply:
I think adding the if-statement is better as in most cases we don't pass a pattern to this tool.

Reply:
ok, here you go